FreeRTOS/FreeRTOS-Plus-TCP
# Check user related headers can be included independently.

Check user related headers can be included independently.

Description
-----------

As #959 mentioned, some header files need to be included in specific order.
This PR solves the header files that needs by user to make sure all headers can be included in any order.
And add a CI test flow to avoid this happening in the future.

Test Steps
-----------

Run new build-combination test.

Checklist:
----------



Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
